File: quide-specification/src/main/java/io/gitlab/arturbosch/quide/platform/processors/ConditionalProcessor.java
package io.gitlab.arturbosch.quide.platform.processors;

import io.gitlab.arturbosch.quide.platform.Processor;
import io.gitlab.arturbosch.quide.platform.UserData;

/**
 * @author Artur Bosch
 */
public interface ConditionalProcessor extends Processor {

	<U extends UserData> boolean isActive(U data);

	@Override
	default <U extends UserData> void execute(U data) {
		if (isActive(data)) {
			doIfActive(data);
		}
	}

	<U extends UserData> void doIfActive(U data);

}

File: quide-api/src/main/kotlin/io.gitlab.arturbosch.quide.api/processors/ConditionalProcessor.kt
package io.gitlab.arturbosch.quide.api.processors

import io.gitlab.arturbosch.quide.api.AnalysisContext

/**
 * @author Artur Bosch
 */
interface ConditionalProcessor : Processor {

	fun isActive(context: AnalysisContext): Boolean

	override fun execute(context: AnalysisContext) {
		if (isActive(context)) {
			onExecution(context)
		}
	}

	fun onExecution(context: AnalysisContext)

}